assembl/tests/model_tests/test_discussion.py

```diff
@@ -16,3 +16,107 @@ def test_add_discussion(test_session):
     assert d.discussion_locales == ['en', 'fr', 'de']
     assert d.terms_and_conditions.entries[0].value == u"Use the haptic JSON system, then you can quantify the cross-platform capacitor!"  # noqa: E501
     assert d.legal_notice.entries[0].value == u"Use the optical SCSI microchip, then you can generate the cross-platform pixel!"  # noqa: E501
+
+
+def test_post_is_read(discussion, post_viewed, post_viewed2,
+                      participant1_user, participant2_user, root_post_1):
+    user_id = participant1_user.id
+    user_id2 = participant2_user.id
+    read_post_ids = discussion.read_post_ids(user_id)
+    read_post_ids2 = discussion.read_post_ids(user_id2)
+    ids = list(read_post_ids)
+    ids2 = list(read_post_ids2)
+    assert len(ids) == 1
+    assert len(ids2) == 1
+    assert ids[0] == root_post_1.id
+    assert ids2[0] == root_post_1.id
+
+
+def test_post_is_unread(discussion,
+                        participant1_user, root_post_1):
+    user_id = participant1_user.id
+    read_post_ids = discussion.read_post_ids(user_id)
+    ids = list(read_post_ids)
+    assert len(ids) == 0
+
+
+def test_post_is_read_by_participant2_user(discussion, post_viewed,
+                                           participant2_user, root_post_1_with_positive_message_classifier):
+    user_id = participant2_user.id
+    read_post_ids = discussion.read_post_ids(user_id)
+    ids = list(read_post_ids)
+    assert len(ids) == 0
+
+
+def test_get_user_permissions(discussion_with_permissions, participant1_user):
+    user_permissions = discussion_with_permissions.get_user_permissions(
+        participant1_user.id)
+    assert sorted(user_permissions) == sorted(
+        [
+            u'add_post',
+            u'delete_my_post',
+            u'edit_my_post',
+            u'read',
+            u'read_public_cif',
+            u'self_register',
+            u'vote'])
+
+
+def test_get_top_ideas(discussion,
+                       root_idea, idea_with_en_fr, subidea_1, subidea_1_1,
+                       subidea_1_1_1, subidea_1_2, announcement_en_fr):
+    top_ideas = discussion.get_top_ideas()
+    for idea in [root_idea, idea_with_en_fr]:
+        assert idea in top_ideas
+    for idea in [subidea_1, subidea_1_1, subidea_1_1_1,
+                 subidea_1_2, announcement_en_fr]:
+        assert idea not in top_ideas
+
+
+def test_count_post_viewers1(discussion, root_post_1, post_viewed):
+    number_of_post_viewers = discussion.count_post_viewers()
+    assert number_of_post_viewers == 1
+
+
+def test_count_post_viewers2(
+        discussion, root_post_1, post_viewed, post_viewed2):
+    number_of_post_viewers = discussion.count_post_viewers()
+    assert number_of_post_viewers == 2
+
+
+def test_get_discussion_urls(discussion):
+    discussion_urls = discussion.get_discussion_urls()
+    assert 'http://localhost:6546/jacklayton2' == discussion_urls[0]
```

Comment:
You shouldn't hard code the domain here, it may change. You can use:
`assert discussion_urls[0].endswith('/jacklayton2')`